Sources/InteroperabilityTests/TestProvider.swift

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import GRPCCore
+
+@available(macOS 13.0, *)
+public struct TestProvider: Grpc_Testing_TestService.ServiceProtocol {
+  public func unimplementedCall(
+    request: GRPCCore.ServerRequest.Single<Grpc_Testing_TestService.Method.UnimplementedCall.Input>
+  ) async throws
+    -> GRPCCore.ServerResponse.Single<Grpc_Testing_TestService.Method.UnimplementedCall.Output>
+  {
+    throw RPCError(code: .unimplemented, message: "The RPC is not implemented.")
+  }
+
+  /// Server implements `emptyCall` which immediately returns the empty message.
+  public func emptyCall(
+    request: GRPCCore.ServerRequest.Single<Grpc_Testing_TestService.Method.EmptyCall.Input>
+  ) async throws -> GRPCCore.ServerResponse.Single<Grpc_Testing_TestService.Method.EmptyCall.Output>
+  {
+    let message = Grpc_Testing_TestService.Method.EmptyCall.Output()
+    return GRPCCore.ServerResponse.Single(message: message)
+  }
+
+  /// Server implements `unaryCall` which immediately returns a `SimpleResponse` with a payload
+  /// body of size `SimpleRequest.responseSize` bytes and type as appropriate for the
+  /// `SimpleRequest.responseType`.
+  ///
+  /// If the server does not support the `responseType`, then it should fail the RPC with
+  /// `INVALID_ARGUMENT`.
+  public func unaryCall(
+    request: GRPCCore.ServerRequest.Single<Grpc_Testing_TestService.Method.UnaryCall.Input>
+  ) async throws -> GRPCCore.ServerResponse.Single<Grpc_Testing_TestService.Method.UnaryCall.Output>
+  {
+    if request.message.responseStatus.isInitialized {
+      let code = Status.Code(rawValue: Int(request.message.responseStatus.code))
+      let status = Status.init(
+        code: code ?? .unknown,
```

Comment:
If `code` is `nil` then the request is ill-formed so we should throw an error with `invalidArgument` and an appropriate message